slsim/Halos/halos_lens_base.py

```diff
@@ -496,6 +496,9 @@ def _build_lens_data(self, halos, mass_correction, z1, z2):
         # If this above code need to be changed, notice the change in the following code
         # including the lens_cosmo_dict one since it assume halos is in front of mass sheet
         if not combined_redshift_list.size:
+            warnings.warn(
```

Comment:
it looks like you put all these warnings back. Especially this one should not appear

Reply:
I synchronized with the TZ's changes.